### PR TITLE
Fix/tile iplb home

### PR DESCRIPTION
**Pull request by varun257**
### Requirements

Migrate tile in load balancer home to oui kit standards

### Description of the Change

replace cui tile with oui tile component
refactor code to move logic to controller 
remove dependency on cui-tile

### Benefits

Brings iplb home page to oui kit latest standards

### Possible Drawbacks

None

### Applicable Issues
SCDC-860

### CODE REVIEW

File: client/app/iplb/home/iplb-home.controller.js [hunk 1]
```diff
@@ -237,6 +235,21 @@ class IpLoadBalancerHomeCtrl {
...
+            this.detailedRegions = !_.isArray(regions) ?
+                [this.RegionService.getRegion(regions)] :
+                _.map(regions, region => this.RegionService.getRegion(region));
+        }
+
+        this.regionsGroup = _.groupBy(this.detailedRegions, "country");
+    }
+
+    hasMultipleRegions () {
+        return this.detailedRegions && this.detailedRegions.length > 1;
```
Comment (FredericEspiau) [inline client/app/iplb/home/iplb-home.controller.js]: `_(this.detailedRegions).isArray() && !_(this.detailedRegions).isEmpty()`

File: client/app/iplb/home/iplb-home.html [hunk 2]
```diff
@@ -1,70 +1,93 @@
...
+                        <oui-skeleton data-randomized></oui-skeleton>
+                        <oui-skeleton data-randomized></oui-skeleton>
+                    </oui-tile-description>
+                </oui-tile-definition>
+                <div ng-if="!ctrl.usage.loading">
+                    <oui-tile-definition data-ng-repeat="quota in (ctrl.usage.data | orderBy:'zone') track by $index" >
+                        <oui-tile-description class="iplb-usage__content">
                             <div class="iplb-usage__region">
-                                <i class="flag-icon {{quota.region.icon}} flag flag__icon-sm cui-tile-definition-description-region__flag" data-ng-class="{ 'cui-tile-definition-description-region__flag_last': $last }"></i>
+                                <i class="flag-icon {{quota.region.icon}} flag flag__icon-sm mr-1"></i>
```
Comment (FredericEspiau) [inline client/app/iplb/home/iplb-home.html]: `aria-hidden="true"`

File: client/app/iplb/home/iplb-home.html [hunk 3]
```diff
@@ -1,70 +1,93 @@
...
         </div>
         <div class="col-xm-12 col-md-4">
-             <div class="cui-tile tile-iplb-usage h-100">
-                <h4 class="oui-header_4 cui-tile__title" data-translate="iplb_home_tile_usage"></h4>
-                <div class="cui-tile__body cui-tile__body_custom cui-tile__body_custom-spaced">
-                    <div class="iplb-usage cui-dropdown-menu-container"
-                        data-ng-repeat="quota in (ctrl.usage.data | orderBy:'zone') track by $index">
-                        <div class="iplb-usage__content">
+            <oui-tile data-heading="{{ ::'iplb_home_tile_usage' | translate }}" class="oui-tile-h-100">
+                <oui-tile-definition ng-if="ctrl.usage.loading">
```
Comment (JeremyDec) [inline client/app/iplb/home/iplb-home.html]: data-

File: client/app/iplb/home/iplb-home.html [hunk 4]
```diff
@@ -1,70 +1,93 @@
...
+                        <oui-skeleton data-randomized></oui-skeleton>
+                        <oui-skeleton data-randomized></oui-skeleton>
+                        <oui-skeleton data-randomized></oui-skeleton>
+                        <oui-skeleton data-randomized></oui-skeleton>
+                        <oui-skeleton data-randomized></oui-skeleton>
+                        <oui-skeleton data-randomized></oui-skeleton>
+                        <oui-skeleton data-randomized></oui-skeleton>
+                    </oui-tile-description>
+                </oui-tile-definition>
+                <div ng-if="!ctrl.usage.loading">
```
Comment (JeremyDec) [inline client/app/iplb/home/iplb-home.html]: `data-`

File: client/app/iplb/home/iplb-home.html [hunk 5]
```diff
@@ -82,82 +105,183 @@ <h4 class="oui-header_4 cui-tile__title">
...
+                    <oui-tile-definition data-term="{{ ::'iplb_home_tile_configuration_private_network' | translate }}">
+                        <oui-tile-description>
+                            <span data-translate="{{ 'iplb_vrack_box_status_' + ctrl.vrackCreationRules.data.status }}"></span>
+                            <small data-ng-if="ctrl.vrackCreationRules.data.status === 'active'">({{ ctrl.vrackCreationRules.data.displayName }})</small>
+                        </oui-tile-description>
+                    </oui-tile-definition>
+                    <oui-tile-definition data-term="{{ ::'iplb_home_tile_configuration_quota_datacenter' | translate }}">
+                        <oui-tile-description>
+                            <div data-ng-if="!ctrl.hasMultipleRegions()">
+                                <i class="flag-icon {{ ctrl.detailedRegions[0].icon }} flag flag__icon-sm mr-1"></i>
```
Comment (FredericEspiau) [inline client/app/iplb/home/iplb-home.html]: `aria-hidden="true"`, also `::`

File: client/app/iplb/home/iplb-home.html [hunk 6]
```diff
@@ -82,82 +105,183 @@ <h4 class="oui-header_4 cui-tile__title">
...
+                                <span>
+                                    <span data-ng-bind=":: ctrl.detailedRegions[0].microRegion.text"></span> -
```
Comment (FredericEspiau) [inline client/app/iplb/home/iplb-home.html]: 👍

File: client/app/iplb/home/iplb-home.html [hunk 7]
```diff
@@ -82,82 +105,183 @@ <h4 class="oui-header_4 cui-tile__title">
...
+                                    <small data-ng-bind=":: ctrl.detailedRegions[0].country"></small>
+                                </span>
+                            </div>
+                            <div data-ng-if="ctrl.hasMultipleRegions()" data-ng-repeat="(key, regions) in ctrl.regionsGroup track by $index">
+                                <i class="flag-icon {{ regions[0].icon }} flag flag__icon-sm mr-1"></i>
```
Comment (FredericEspiau) [inline client/app/iplb/home/iplb-home.html]: `aria-hidden="true"`, also `::`

File: client/app/iplb/home/iplb-home.html [hunk 8]
```diff
@@ -82,82 +105,183 @@ <h4 class="oui-header_4 cui-tile__title">
...
+                                <span>
```
Comment (FredericEspiau) [inline client/app/iplb/home/iplb-home.html]: Useless `span`

File: client/app/oui-less-override.less [hunk 9]
```diff
@@ -68,3 +68,10 @@
         border-left-width: 1px;
     }
 }
+
+.oui-tile-h-100 {
+    // This fix can be removed once fixed in the component.
+    .oui-tile {
```
[the remark this answers is not in the document]
Comment (varun257) [reply client/app/oui-less-override.less]: Nope, h-100 adds property on tile component. Component has a <div class="oui-tile"> where we want to override the class.

[the remark this answers is not in the document]
Comment (jleveugle) [reply client/app/oui-less-override.less]: @varun257, why do you need to override the style?

We are creating UI-Kit to standardize our styles and components, if you need to change it, we should discuss about that to understand what is needed and why our components doesn't fit with our needs :)

cc @AxelPeter :)

**Comment by varun257** [comment]
@FredericEspiau @marie-j Have taken care of all listed changes. Please have a look.

File: client/app/iplb/home/iplb-home.controller.js [hunk 10]
```diff
@@ -248,7 +248,8 @@ class IpLoadBalancerHomeCtrl {
     }
 
     hasMultipleRegions () {
-        return this.detailedRegions && this.detailedRegions.length > 1;
+        return _(this.detailedRegions).isArray() && !_(this.detailedRegions).isEmpty();
+        // return this.detailedRegions && this.detailedRegions.length > 1;
```
Comment (FredericEspiau) [inline client/app/iplb/home/iplb-home.controller.js]: Useless comment

File: client/app/oui-less-override.less (hunk 9 above)
[the remark this answers is not in the document]
Comment (varun257) [reply client/app/oui-less-override.less]: @jleveugle We have created a bug for UI KIT team and they are working on a fix. This is just a temporary fix and will be removed as soon as we get a new release.

File: client/app/iplb/home/iplb-home.controller.js [hunk 11]
```diff
@@ -248,7 +248,7 @@ class IpLoadBalancerHomeCtrl {
     }
 
     hasMultipleRegions () {
-        return _(this.detailedRegions).isArray() && !_(this.detailedRegions).isEmpty();
+        return _(this.detailedRegions).isArray() && this.detailedRegions.length > 1;
```
Comment (FredericEspiau) [inline client/app/iplb/home/iplb-home.controller.js]: Indeed, sorry for misguiding you 😄

Comment (varun257) [reply client/app/iplb/home/iplb-home.controller.js]: :)

**Comment by ovh-cds** [comment]
CDS Report build#5323.0 ✘  
* Stage 1
  * deps ✘

**Comment by ovh-cds** [comment]
CDS Report build#5323.1 ✘  
* Stage 1
  * deps ✘

**Comment by ovh-cds** [comment]
CDS Report build#5323.2 ✘  
* Stage 1
  * deps ✘

**Comment by ovh-cds** [comment]
CDS Report build#5323.3 ✘  
* Stage 1
  * deps ✘

**Pull request merged**